lib/api/objectCopy.js

```diff
@@ -351,6 +352,29 @@ function objectCopy(authInfo, request, sourceBucket,
                     serverSideEncryption, destBucketMD);
             }
 
+            if (storeMetadataParams.locationMatch && locationType &&
+            constants.externalBackends[locationType]
+            && dataLocator.length === 1) {
```

Comment:
would the length ever be greater than 1? are you just checking to make sure there is any data at all?

Reply:
Length should always be 1 for external dataLocator.